### PR TITLE
Checkpoint parser improvements

### PR DESCRIPTION
**Pull request by bazsi**
With new, current samples received from a syslog-ng user, I could improve the parser to better support the output of CheckPoint LogExporter. Also added links to official documentation and community topic on the matter.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: tests/python_functional/functional_tests/parsers/app-parser/test_topic_syslog_raw.py [hunk 1]
```diff
@@ -30,6 +30,8 @@
...
+    (r"""<134>1 2018-03-21T17:25:25 MDS-72 CheckPoint 13752 - [action:"Update"; flags:"150784"; ifdir:"inbound"; logid:"160571424"; loguid:"{0x5ab27965,0x0,0x5b20a8c0,0x7d5707b6}"; origin:"192.168.32.91"; originsicname:"CN=GW91,O=Domain2_Server..cuggd3"; sequencenum:"1"; time:"1521645925"; version:"5"; auth_method:"Machine Authentication (Active Directory)"; auth_status:"Successful Login"; authentication_trial:"this is a reauthentication for session 9a026bba"; client_name:"Active Directory Query"; client_version:"R80.10"; domain_name:"spec.mgmt"; endpoint_ip:"192.168.32.69"; identity_src:"AD Query"; identity_type:"machine"; product:"Identity Awareness"; snid:"9a026bba"; src:"192.168.32.69"; src_machine_group:"All Machines"; src_machine_name:"yonatanad";]""", "${.app.name}", "checkpoint"),
```
Comment (mitzkia) [inline tests/python_functional/functional_tests/parsers/app-parser/test_topic_syslog_raw.py]: Sorry, as I see this line (33) is the same as the previous (32).

Comment (mitzkia) [reply tests/python_functional/functional_tests/parsers/app-parser/test_topic_syslog_raw.py]: Sorry, they are not the same.

**Comment by mitzkia** [comment]
I have checked Light testcases, at first glance I have found a new testcase which was almost the same as an old one, but after checked it again they are different in timestamps. 
Light testcases are ok.

**Comment by jewnix** [comment]
I'm running version R80.20m2 and sending it syslog format from CP, and it does not seem to parse it correctly. maybe related to the `date-parser`

actual output:
`Jul 29 13:04:13 checkpoint <134>1 2019-07-29T17:04:13Z checkpoint CheckPoint 9951 - [action:"Accept"; conn_direction:"Outgoing"; contextnum:"1"; flags:"7258112";`

**Comment by bazsi** [comment]
Can you show your config?

**Comment by jewnix** [comment]
> filter f_cpdash {
    netmask("10.0.0.1/32")  or host('cpdash-swg');
};
destination d_cpdash {
    file (
      "/var/log/splunk/cp_firewall/$LOGHOST/$HOST/$R_YEAR$R_MONTH$R_DAY-$R_HOUR.log");
};
log {
    source {
        network(
        transport("tcp")
        flags(no-parse)
        port(2154)
        );
    };
    parser { checkpoint-parser(); };
    filter(f_cpdash);
    destination(d_cpdash);
    flags(final);
};

**Comment by bazsi** [comment]
I wanted to check how you apply the checkpoint parser and it seems fine.

If you start syslog-ng in debug and trace enabled (like with -Fedvt command
line options from a shell), you can see how checkpoint parser attempts
parsing the message and where it fails. If you have problems deciphering
the output just post it here.

**Pull request merged**